src/business-logic/services/device-service-implementation.ts

```diff
@@ -0,0 +1,25 @@
+import { DeviceService } from './interfaces/device-service'
+import { Device } from '../../model/entities/device'
+import { DeviceRepository } from '../../data-access/repositories/interfaces/device-repository'
+import { DeviceEventEmitter } from './interfaces/device-event-emitter'
+
+export class DeviceServiceImplementation implements DeviceService {
+  constructor(private readonly deviceRepository: DeviceRepository, private readonly deviceEventEmitter: DeviceEventEmitter) {}
+  
+  public async getDevices(): Promise<Device[]> {
+    return await this.deviceRepository.getDevices()
+  }
+
+  public async getDevice(deviceId: string): Promise<Device> {
+    const deviceConfig = await this.deviceRepository.getDevice(deviceId)
```

Comment:
Since our entity is called `Device` it would be nice that this variable also was called `device`.
Also, please specify the return type. Your if-statement on the next line indicates that the type is `Device | undefined`. But `deviceRepository.getDevice` returns `Promise<Device>` so the the variable can never be undefined. If you had included the type, then you would have seen that your if-statement is unnecesary :)